content/en/blog/_posts/2025-08-27-Kubernetes-v1-34-Release/index.md

```diff
@@ -0,0 +1,509 @@
+
+To address this visibility gap, distributed tracing capabilities are introduced directly into the `kubelet` using the OpenTelemetry standard. By enabling this feature, operators can export detailed trace data to their observability backends. This provides an unprecedented level of insight, allowing for the precise tracing of requests related to the Pod lifecycle, volume management, and other critical functions, which is essential for effective debugging and performance analysis.
+
+This work was done as part of [KEP \#2831](https://kep.k8s.io/2831) led by SIG Node.
+
+### Allow special characters in environment variables
+
+The environment variable validation rules in Kubernetes have been relaxed to allow nearly all printable ASCII characters, excluding \=. This change supports scenarios where workloads require nonstandard characters in variable names \- for example, frameworks like .NET Core that use :  to represent nested config paths.
+
+The relaxed validation applies to environment variables defined directly in Pod spec, as well as those injected using envFrom references to ConfigMaps and Secrets.
+
+This work was done as part of [KEP \#4369](https://kep.k8s.io/4369) led by SIG Node.
+
+### Taint management is separated from Node lifecycle
+
+Historically, the `TaintManager`'s logic for applying NoSchedule and NoExecute taints to nodes based on their condition (e.g., NotReady, Unreachable) was tightly coupled with the `NodeLifecycleController`. This tight coupling made the code harder to maintain and test, and it also limited the flexibility of the taint-based eviction mechanism.  
+This KEP refactors the `TaintManager` into its own separate controller within the Kubernetes controller manager. It is an internal architectural improvement designed to increase code modularity and maintainability. This change allows the logic for taint-based evictions to be tested and evolved independently, but it has no direct user-facing impact on how taints are used.
+
+This work was done as part of [KEP \#3902](https://kep.k8s.io/3902) led by SIG Node.
+
+## New features in Beta
+
+*This is a selection of some of the improvements that are now beta following the v1.34 release.*
+
+### `.kuberc` file for `kubectl` user preferences
+
+A `.kuberc` configuration file allows you to define preferences for `kubectl`, such as default options and command aliases. Unlike the kubeconfig file, the `.kuberc` configuration file does not contain cluster details, usernames or passwords.  
+This feature was introduced as alpha in v1.33, gated behind the environment variable `KUBECTL_KUBERC`. It has graduated to beta in v1.34 and is enabled by default.
+
+This work was done as part of [KEP \#3104](https://kep.k8s.io/3104) led by SIG CLI.
+
+### External ServiceAccount token signing
+
+\<TODO 1-2 PARAGRAPH DESCRIPTION OF CHANGE\>
```

Comment:
```suggestion
Traditionally, Kubernetes manages ServiceAccount tokens using static signing keys that are loaded from disk at `kube-apiserver` startup. This feature introduces an `ExternalJWTSigner` gRPC service for out-of-process signing, enabling Kubernetes distributions to integrate with external key management solutions (for example, HSMs, cloud KMSes) for ServiceAccount token signing instead of static disk-based keys.

Introduced as alpha in v1.32, this external JWT signing capability advances to beta and is enabled by default in v1.34.
```